swarm/storage/dbstore.go

```diff
@@ -451,6 +478,7 @@ func (s *DbStore) getEntryCnt() uint64 {
 }
 
 func (s *DbStore) Close() {
+	close(s.batchesC)
```

Comment:
To confirm this, either run the tests with the race detector on, or just eyeball the code :)

To fix it, you usually have two channels: one gets closed here, and the writing goroutine selects on it and then stops writing to the other channel.